### DISCUSSION TITLE
Navigation Bar that persist accross multiple pages

### DISCUSSION THREAD

**Discussion by 1Miguel** [zauberzeug/nicegui#4070] [Q&A] [answered]
### Question

Hello!

I created a navigation bar that routes to different pages when clicked. Any way for it to persist in multiple pages without always adding navigation bar code to each ui page? or should I just use a decorator?

![image](https://github.com/user-attachments/assets/9f545f9a-a42d-41e6-9965-8f10725ea5d6)

**Comment by rodja** [comment, marked as answer]
Have a look at our [modularization example](https://github.com/zauberzeug/nicegui/tree/main/examples/modularization).
